lib/liquid/registers/disabled_tags.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+module Liquid
+  class DisabledTags < Register
+    def initialize
+      @disabled_tags = Hash.new { |h, k| h[k] = 0 }
```

Comment:
I think it may be more performant to not use a hash with default block. That means each time we call `disabled?` with a tag, the proc will have to run to initialize the `0` and assign the key. Instead we could be checking `@disabled_tags.key?(tag)` which would be much faster, at the cost of making `increment` / ` decrement` a 2-liner instead of 1-liner.